Tests/NIOSSLTests/UnwrappingTests.swift

```diff
@@ -547,14 +542,13 @@ final class UnwrappingTests: XCTestCase {
         // of a CLOSE_NOTIFY.
         var buffer = clientChannel.allocator.buffer(capacity: 1024)
         buffer.writeStaticString("GET / HTTP/1.1\r\nHost: localhost\r\nContent-Length: 0\r\n\r\n")
-
-        do {
-            try clientChannel.writeInbound(buffer)
-            XCTFail("Did not error")
-        } catch NIOSSLError.shutdownFailed {
-            // expected
-        } catch {
-            XCTFail("Unexpected error: \(error)")
+        
+        XCTAssertThrowsError(try clientChannel.writeInbound(buffer)) { error in
+            if case .shutdownFailed = error as? NIOSSLError {
+                // Okay
+            } else {
+                XCTFail("Unexpected error: \(error)")
+            }
```

Comment:
This can be a one-line `XCTAssertEqual` as used elsewhere in this patch.

Reply:
Actually I was getting error : 

```
UnwrappingTests.swift:547:13: error: generic parameter 'T' could not be inferred
            XCTAssertEqual(.shutdownFailed, error as? NIOSSLError)
            ^
XCTest.XCTAssertEqual:1:13: note: in call to function 'XCTAssertEqual(_:_:_:file:line:)'
public func XCTAssertEqual<T>(_ expression1: @autoclosure () throws -> T, _ expression2: @autoclosure () throws -> T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) where T : Equatable
            ^

```

Reply:
`.shutdownFailed` has associated data, so you you can't construct it that way. Use `switch`.

Reply:
Done! Please review once.

Reply:
This should still use `XCTAssertThrowsError`: the `switch` can just be in the body.

Reply:
Done!